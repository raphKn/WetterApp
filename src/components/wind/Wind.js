import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View } from 'react-native';
import CustomSection from '../common/CustomSection';
import { isEmptyObject } from '../../utils/ObjectUtils';
import Windrichtung from './WindRichtung';
import WindGeschwindigkeit from './WindGeschwindigkeit';

const styles = StyleSheet.create({
    image: { width: 70, height: 70 },
    temperaturenContainer: {
        flexDirection: 'row',
        justifyContent: 'space-around',
    },
    temperaturenHeader: {
        fontSize: 24,
    },
});

const Wind = (props) => {
    if (isEmptyObject(props.wind)) {
        return null;
    }
    return (<View>
        <CustomSection style={styles.temperaturenContainer} name="Wind">
            <Windrichtung bezeichnung={props.wind.bezeichnung} imageRotation={props.wind.imageRotation} />
            <WindGeschwindigkeit geschwindigkeit={props.wind.geschwindigkeit} />
        </CustomSection>
    </View>);
};

Wind.propTypes = ({
    wind: PropTypes.shape({
        bezeichnung: PropTypes.string,
        geschwindigkeit: PropTypes.number,
        imageRotation: PropTypes.number,
    }),
});

Wind.defaultProps = ({
    wind: null,
});

export default Wind;