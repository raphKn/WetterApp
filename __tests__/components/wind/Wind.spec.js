import React from 'react';
import renderer from 'react-test-renderer';
import Wind from '../../../src/components/wind/Wind';

describe('Wind', () => {
    it('renders strong wind correctly in a snapshot', () => {
        const wind = {
            bezeichnung: 'Ost',
            geschwindigkeit: 5.2,
            imageRotation: 90,
        };
        const wrapper = renderer.create(
            <Wind wind={wind} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders weak wind correctly in a snapshot', () => {
        const wind = {
            geschwindigkeit: 0.2,
        };
        const wrapper = renderer.create(
            <Wind wind={wind} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders nothing for no wind in a snapshot', () => {
        const wrapper = renderer.create(
            <Wind wind={{}} />);
        expect(wrapper).toMatchSnapshot();
    });
});
