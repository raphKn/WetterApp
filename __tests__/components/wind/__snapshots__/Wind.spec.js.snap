// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wind renders nothing for no wind in a snapshot 1`] = `null`;

exports[`Wind renders strong wind correctly in a snapshot 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "paddingTop": 12,
        },
        Object {
          "backgroundColor": "#3770a5",
        },
      ]
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#FAFAFA",
            },
            Object {
              "fontSize": 18,
              "paddingHorizontal": 18,
            },
          ]
        }
      >
        Wind
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#E0E0E0",
            "paddingHorizontal": 18,
            "paddingVertical": 6,
          },
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 6,
          }
        }
      >
        <View
          style={
            Object {
              "height": 38.5,
              "width": 38.5,
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "flex": 1,
                  "height": null,
                  "width": null,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "90deg",
                    },
                  ],
                },
              ]
            }
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          aus 
          Ost
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "fontSize": 32,
              },
            ]
          }
        >
          5.2
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          Meter pro Sekunde
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Wind renders weak wind correctly in a snapshot 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "paddingTop": 12,
        },
        Object {
          "backgroundColor": "#3770a5",
        },
      ]
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#FAFAFA",
            },
            Object {
              "fontSize": 18,
              "paddingHorizontal": 18,
            },
          ]
        }
      >
        Wind
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#E0E0E0",
            "paddingHorizontal": 18,
            "paddingVertical": 6,
          },
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "fontSize": 32,
              },
            ]
          }
        >
          0.2
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          Meter pro Sekunde
        </Text>
      </View>
    </View>
  </View>
</View>
`;
